src/Microsoft.Framework.Runtime/DependencyManagement/LockFile.cs

```diff
@@ -15,18 +15,24 @@ public class LockFile
         public IList<LockFileLibrary> Libraries { get; set; } = new List<LockFileLibrary>();
         public IList<LockFileTarget> Targets { get; set; } = new List<LockFileTarget>();
 
-        public bool IsValidForProject(Project project)
+        public bool IsValidForProject(Project project, out string message)
         {
+            message = null;
+
             if (Version != LockFileReader.Version)
             {
+                message = $"Expected lock file version is {LockFileReader.Version}, while actual version is {Version}";
```

Comment:
How about: `$"The expected lock file version ({LockFileReader.Version}) does not match the actual version ({Version})."` ?